Cilsil/Cil/Parsers/InstructionParser.cs

```diff
@@ -138,10 +138,68 @@ protected void RegisterNode(ProgramState state, CfgNode node)
                              new ConstExpression(ProcedureName.BuiltIn__new),
                              args,
                              callFlags,
-                             state.CurrentLocation), 
+                             state.CurrentLocation),
                     new VarExpression(newObjectIdentifier));
         }
 
+        /// <summary>
+        /// Creates a <see cref="CfgNode"/> containing instructions for memory allocation as well
+        /// as constructor invocation.
+        /// </summary>
+        /// <param name="type">The SIL type to be allocate memory for.</param>
+        /// <param name="state">Current program state.</param>
+        /// <returns>Node with the object allocation instructions, as well as the variable which 
+        /// represents the new object.</returns>
+        protected static (CfgNode, VarExpression) CreateObjectAllocationNode(Tptr type,
+                                                                             ProgramState state)
+        {
+            var typeName = type.StripPointer().ToString();
+
+            var newObjectIdentifier = state.GetIdentifier(Identifier.IdentKind.Normal);
+            var newObjectVariable = new VarExpression(newObjectIdentifier);
+            var callFlags = new Call.CallFlags(isVirtual: false,
+                                               noReturn: false,
+                                               isObjCBlock: false);
+            var objectAllocationCall =
+                new Call(newObjectIdentifier,
+                         type,
+                         new ConstExpression(ProcedureName.BuiltIn__new),
+                         new List<Call.CallArg>
+                         {
+                                     new Call.CallArg(
+                                         new SizeofExpression(
+                                             type.StripPointer(), "exact"),
+                                         type)
+                         },
+                         callFlags,
+                         state.CurrentLocation);
+
+            var objectConstructorCall =
+                new Call(state.GetIdentifier(Identifier.IdentKind.Normal),
+                         new Tvoid(),
+                         new ConstExpression(new ProcedureName(".ctor",
+                                                               new List<string>(),
+                                                               typeName,
+                                                               "System.Void",
+                                                               false)),
+                         new List<Call.CallArg>
+                         {
+                                     new Call.CallArg(newObjectVariable, type)
```

Comment:
nit: too many indentations?

Reply:
I think not? what are you thinking the proper indentation should be?

Reply:
One tab behind the last character of previous line? To me it looks like 4 tabs here.

Reply:
ah you're right -- thanks.